### PR TITLE
Update SRM to 1.3.0-rc3-24005-00, SCI to 1.2.0-rc3-24005-00 and DSR to 1.0.8-rc2-60325.

### PR DESCRIPTION
**Pull request by tmat**

### CODE REVIEW

**Comment by tmat** [comment]
@agocke Please review - the same change ported to future.

File: src/Debugging/Microsoft.DiaSymReader.PortablePdb.Tests/project.json [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
   "dependencies": {
     "Microsoft.NETCore.Portable.Compatibility": "1.0.0",
-    "System.Collections": "4.0.10",
-    "System.Diagnostics.Debug": "4.0.10",
-    "System.Runtime": "4.0.20",
-    "System.Runtime.Extensions": "4.0.10",
-    "System.Runtime.InteropServices": "4.0.20",
-    "System.IO.FileSystem": "4.0.0",
-    "System.Runtime.InteropServices": "4.0.20",
+    "System.Collections": "4.0.11-rc3-24005-00",
```
Comment (agocke) [inline src/Debugging/Microsoft.DiaSymReader.PortablePdb.Tests/project.json]: Will any of these changes cause us to change our insertion payload?

Do we need to insert more facade assemblies now?

File: src/ExpressionEvaluator/Core/Source/Concord/Microsoft.VisualStudio.Debugger.Engine.Portable.il [hunk 2]
```diff
@@ -6,17 +6,17 @@
 .assembly extern System.Globalization
 {
 	.publickeytoken = (B0 3F 5F 7F 11 D5 0A 3A)
-	.ver 4:0:10:0
```
Comment (agocke) [inline src/ExpressionEvaluator/Core/Source/Concord/Microsoft.VisualStudio.Debugger.Engine.Portable.il]: What's this?

File: src/Interactive/CsiCore/project.json [hunk 3]
```diff
@@ -1,5 +1,7 @@
 ﻿{
-  "dependencies": { },
+  "dependencies": {
+    "System.AppContext": "4.1.0-rc3-24005-00"
```
Comment (agocke) [inline src/Interactive/CsiCore/project.json]: Why is this necessary?

File: src/Interactive/CsiCore/Csi.cs [hunk 4]
```diff
@@ -24,6 +24,7 @@ internal static int Main(string[] args)
                     baseDirectory: Directory.GetCurrentDirectory(),
                     sdkDirectoryOpt: CorLightup.Desktop.TryGetRuntimeDirectory(),
                     args: args,
+                    clientDirectory: AppContext.BaseDirectory,
```
Comment (agocke) [inline src/Interactive/CsiCore/Csi.cs]: ? Why does Csi need the client directory now?

File: src/Interactive/csi/csi.csproj [hunk 5]
```diff
@@ -15,6 +15,7 @@
     <SolutionDir Condition="'$(SolutionDir)' == '' OR '$(SolutionDir)' == '*Undefined*'">..\..\..\</SolutionDir>
     <RestorePackages>true</RestorePackages>
     <TargetFrameworkVersion>v4.6</TargetFrameworkVersion>
+    <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
```
Comment (agocke) [inline src/Interactive/csi/csi.csproj]: I'm pretty sure this is wrong.

File: src/Debugging/Microsoft.DiaSymReader.PortablePdb.Tests/project.json (hunk 1 above)
Comment (tmat) [inline src/Debugging/Microsoft.DiaSymReader.PortablePdb.Tests/project.json]: I wouldn't think so. When targeting 4.6 we shouldn't need more facades than what we have already.

File: src/Debugging/Microsoft.DiaSymReader.PortablePdb.Tests/project.json [hunk 6]
```diff
@@ -1,13 +1,12 @@
...
+    "System.Diagnostics.Debug": "4.0.11-rc3-24005-00",
```
Comment (agocke) [inline src/Debugging/Microsoft.DiaSymReader.PortablePdb.Tests/project.json]: Are all of these upgrades necessary? Is there anything we can keep at RTM?

File: src/ExpressionEvaluator/Core/Source/Concord/Microsoft.VisualStudio.Debugger.Engine.Portable.il (hunk 2 above)
Comment (tmat) [inline src/ExpressionEvaluator/Core/Source/Concord/Microsoft.VisualStudio.Debugger.Engine.Portable.il]: These are reference assemblies taken from the debugger. Turns out that the debugger is targeting assemblies incompatible with Profile7. Will need to address that somehow later.

File: src/Interactive/CsiCore/Csi.cs (hunk 4 above)
Comment (tmat) [inline src/Interactive/CsiCore/Csi.cs]: This is just refactoring - this was moved up from the constructor.

File: src/Interactive/CsiCore/project.json (hunk 3 above)
Comment (tmat) [inline src/Interactive/CsiCore/project.json]: "Workaround for duplicate AppContext (dotnet#10309)"

File: src/Interactive/csi/csi.csproj (hunk 5 above)
Comment (tmat) [inline src/Interactive/csi/csi.csproj]: Why?

**Comment by tmat** [comment]
eta failure is expected. Roslyn internal change needs to be checked in as well.

File: src/Debugging/Microsoft.DiaSymReader.PortablePdb.Tests/project.json (hunk 6 above)
Comment (tmat) [inline src/Debugging/Microsoft.DiaSymReader.PortablePdb.Tests/project.json]: Yes, all these are referenced by Immutable Collections.

File: src/Interactive/csi/csi.csproj (hunk 5 above)
Comment (tmat) [inline src/Interactive/csi/csi.csproj]: Removed.

**Comment by tmat** [comment]
@dotnet-bot retest prtest/win/dbg/eta please
// Previous failure: http://dotnet-ci.cloudapp.net/job/dotnet_roslyn_prtest_win_dbg_eta/5851/
// Retest reason:
